### PR TITLE
Release note 177 RC1

### PR DESCRIPTION
**Pull request by marionf**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Release note 177 RC1

### CODE REVIEW

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+![1.7.7.0 RC1 is available!](/assets/images/2020/10/build-177RC1.png)
+
+This release candidate aims at making sure that all major regressions found during the beta have been fixed, and that no new ones have been spotted. If no new critical or major regression is reported by November 6th, we will make the final version of 1.7.7.0 available.
+
+Please note that it is not currently possible to perform an automatic upgrade from the Beta to the Release Candidate. Furthermore, it will not be possible to upgrade from this Release Candidate to the final release. If you perform an update of your test store to the RC version, you will need to rollback to your current version before being able to upgrade to the final 1.7.7.0.
+
+Please download, install and test this new version of PrestaShop and give us your feedback as early as possible. The later a problem is identified, the later it can be fixed and this delays the final output.
+If you find regressions compared to the 1.7.6 version, please open an issue and describe it on [Github](https://github.com/PrestaShop/PrestaShop/issues/new?template=1_bug_report.md) so that it can be reproduced and fixed as soon as possible.
```
Comment (matks) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: If you do not insert blank empty lines between statements, it will be rendered as a single block (see https://github.com/PrestaShop/prestashop.github.io/blob/07f6aa7f430024c759427619a312556880d36bad/news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md) ; is that intended ?

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- [Missing gift wrapping and recycled paper options in create order page](https://github.com/PrestaShop/PrestaShop/issues/21300)
+- [In order detail page, when adding a discount amount with a different currency than the default one, an incorrect value is displayed and an incorrect amount is calculated](https://github.com/PrestaShop/PrestaShop/issues/21323)
+- [An exception is displayed when removing a discount with free gift from an order](https://github.com/PrestaShop/PrestaShop/issues/21438)
+
+**Back Office:**
+- [Ranges are asked while they shouldn't during carrier edition](https://github.com/PrestaShop/PrestaShop/issues/20993)
+- [Following migration, $_GET identifier parameters are missing on "View {..} Page](https://github.com/PrestaShop/PrestaShop/issues/20934)
+- [Features page - an exception is displayed](https://github.com/PrestaShop/PrestaShop/issues/20777)
+- [Stats page - No module has been installed](https://github.com/PrestaShop/PrestaShop/issues/21186)
+- [An error is displayed when disabling or enabling maintenance
```
Comment (matks) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Warning, the `]` went on the next line

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+**Front Office:**
+- [Wrong cart rules discount when using alternative currency](https://github.com/PrestaShop/PrestaShop/issues/20594)
+- [Incorrect total with Free gift + Percent discount cart rule](https://github.com/PrestaShop/PrestaShop/issues/20690)
+- [When we have two currencies enabled - the Currency block and the language block are not working anymore](https://github.com/PrestaShop/PrestaShop/issues/20950)
+
+## Changelog
+
+90 pull requests have been merged since the beta 2 release and 40+ issues have been fixed. Read the [Changelog](https://github.com/PrestaShop/PrestaShop/releases) for details. 
+This brings us to more than 1300 merged pull requests for the 1.7.7.0 milestone, it is definitely our biggest yet.
```
Comment (matks) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: 💪

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+Internal and external contributors who submitted pull requests and participated to build this release candidate: [PululuK](https://github.com/PululuK), [okom3pom](https://github.com/okom3pom), [ks129](https://github.com/ks129), [Boubker Bribri](https://github.com/boubkerbribri), [Franck Lefèvre](https://github.com/Progi1984), [Ibrahima Sow](https://github.com/sowbiba), [Jonathan Lelievre](https://github.com/jolelievre), [Mathieu Ferment](https://github.com/matks), [Matthieu Rolland](https://github.com/matthieu-rolland), [Nesrine Abdmouleh](https://github.com/nesrineabdmouleh), [Pablo Borowicz](https://github.com/eternoendless), [Pierre Rambaud](https://github.com/PierreRambaud), [Simon Garny](https://github.com/SimonGrn), [Stephane Decisy](https://github.com/SD1982), [Thomas Baccelli](https://github.com/atomiix) and [Valentin Szczupak](https://github.com/NeOMakinG). Thank you!
+
+External contributors who reported regressions to build a more stable 1.7.7: [ks129](https://github.com/ks129), [okom3pom](https://github.com/okom3pom), [MathiasReker](https://github.com/MathiasReker) and [kpodemski](https://github.com/kpodemski). Thank you!
```
Comment (matks) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Well, @kpodemski is no more an external contributor 😛 he's now a maintainer

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md (hunk 1 above)
Comment (marionf) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: No, thank you I am going to fix that !

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md (hunk 2 above)
Comment (marionf) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Nice catch, thanks !

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+![1.7.7.0 RC1 is available!](/assets/images/2020/10/build-177RC1.png)
+
+This release candidate aims at making sure that all major regressions found during the beta have been fixed, and that no new ones have been spotted. If no new critical or major regression is reported by November 6th, we will make the final version of 1.7.7.0 available.
+
+Please note that it is not currently possible to perform an automatic upgrade from the Beta to the Release Candidate. 
+
+Furthermore, it will not be possible to upgrade from this Release Candidate to the final release. If you perform an update of your test store to the RC version, you will need to rollback to your current version before being able to upgrade to the final 1.7.7.0.
+
+Please download, install and test this new version of PrestaShop and give us your feedback as early as possible. The later a problem is identified, the later it can be fixed and this delays the final output.
```
Comment (eternoendless) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]:
```suggestion
Please download, install and test this new version of PrestaShop and give us your feedback as early as possible. The later a problem is identified, the more time it takes to be fixed, and this delays the final release.
```

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md (hunk 1 above)
Comment (ttoine) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: if you need the equivalent of a `< br/>`, in markdown, add 2 spaces at the end of the line.
https://daringfireball.net/projects/markdown/syntax#p

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+If you find regressions compared to the 1.7.6 version, please open an issue and describe it on [Github](https://github.com/PrestaShop/PrestaShop/issues/new?template=1_bug_report.md) so that it can be reproduced and fixed as soon as possible.
+
+**Remember, this release candidate version is pre release software. Do not use it on your production shop!**
+
+
+## Notable changes
```
Comment (ttoine) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: compared to what? the Beta ?
```suggestion
## Notable changes since the Beta release
```

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+
+**Front Office:**
+- [Wrong cart rules discount when using alternative currency](https://github.com/PrestaShop/PrestaShop/issues/20594)
+- [Incorrect total with Free gift + Percent discount cart rule](https://github.com/PrestaShop/PrestaShop/issues/20690)
+- [When we have two currencies enabled - the Currency block and the language block are not working anymore](https://github.com/PrestaShop/PrestaShop/issues/20950)
+
+## Changelog
+
+90 pull requests have been merged since the beta 2 release and 40+ issues have been fixed. Read the [Changelog](https://github.com/PrestaShop/PrestaShop/releases) for details.
```
Comment (jolelievre) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]:
```suggestion
151 pull requests have been merged since the beta 2 release and 40+ issues have been fixed. Read the [Changelog](https://github.com/PrestaShop/PrestaShop/releases) for details. 
```

Comment (jolelievre) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: I don't know about the issues number, but it surely have increased as well
Do you need me to search for the number or do you have a way to find it?

Comment (jolelievre) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: I think it could be nice to add a report about the number of tests added (I'm comparing them)

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+## How to help finish the 1.7.7
+
+Your feedback on this RC is essential, the regressions you find right now are as many bugs less on the final version and less patch versions. It also means fewer problems on your (or your customer’s) online store. 
+
+You must:
+
+- Install it and test that there is no regression compared to the 1.7.6 version
+- If you develop modules or themes, test them on this version
+- Test the auto-upgrade plugin
```
Comment (jolelievre) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Maybe not this one since it's only a RC and it block the upgrade to the final release?
Or should the upgrade still be tested in a preprod environment? (maybe this should be detailed then, even though you mentioned it earlier in the article)

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+- [Stats page - No module has been installed](https://github.com/PrestaShop/PrestaShop/issues/21186)
+- [An error is displayed when disabling or enabling maintenance](https://github.com/PrestaShop/PrestaShop/issues/21430)
+- [Hook displayBackOfficeTop not being called on Login BO Page](https://github.com/PrestaShop/PrestaShop/issues/21379)
+
+
+**Front Office:**
+- [Wrong cart rules discount when using alternative currency](https://github.com/PrestaShop/PrestaShop/issues/20594)
+- [Incorrect total with Free gift + Percent discount cart rule](https://github.com/PrestaShop/PrestaShop/issues/20690)
+- [When we have two currencies enabled - the Currency block and the language block are not working anymore](https://github.com/PrestaShop/PrestaShop/issues/20950)
+
```
[the remark this answers is not in the document]
Comment (marionf) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Done here: https://github.com/PrestaShop/prestashop.github.io/pull/788/commits/eae0784b32ed2a1fee7d0b7ee0dee75da9de93ff

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+**For developers**
+- [Make PrestaShop compatible with composer 2.0](https://github.com/PrestaShop/PrestaShop/pull/21609)
+- [Update behat and gerkhin to allow debugging in PHPStorm](https://github.com/PrestaShop/PrestaShop/pull/21532)
+
+## Changelog
+
+151 pull requests have been merged since the beta 2 release and 40+ issues have been fixed. Read the [Changelog](https://github.com/PrestaShop/PrestaShop/releases) for details. 
+This brings us to more than 1300 merged pull requests for the 1.7.7.0 milestone, it is definitely our biggest yet.
```
Comment (jolelievre) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: I didin't update the total number of PRs but it probably increase as well

Comment (jolelievre) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Yep, 1405 https://github.com/PrestaShop/PrestaShop/pulls?q=is%3Apr+is%3Aclosed+milestone%3A1.7.7.0

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title:  "PrestaShop 1.7.7.0 RC1 Is Ready To Be Tested!"
+subtitle: "A month after the beta, here it is: the first release candidate for PrestaShop 1.7.7.0 is now ready for you to test!"
+date:   2020-11-03 15:00:00
```
Comment (jolelievre) [inline news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Does the file need to be renamed? I believe it is used for the sorting no? and maybe the url?

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md (hunk 4 above)
Comment (ttoine) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: how do we handle that?
I mean, he is still not a PrestaShop employee or a subcontractor... And, historically, the "by / thank you" was made to make a difference.
Then, should this change, and starting for now, we use that to make a difference between maintainers and contributors instead ?
what about PrestaShop employees who are not maintainers, like QA and PM?

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+- [Make PrestaShop compatible with composer 2.0](https://github.com/PrestaShop/PrestaShop/pull/21609)
+- [Update behat and gerkhin to allow debugging in PHPStorm](https://github.com/PrestaShop/PrestaShop/pull/21532)
+
+## Changelog
+
+151 pull requests have been merged since the beta 2 release and 30+ issues have been fixed. Read the [Changelog](https://github.com/PrestaShop/PrestaShop/releases) for details. 
+This brings us to more than 1400 merged pull requests for the 1.7.7.0 milestone, it is definitely our biggest yet.
+
+If you are looking for more details about all changes and new features expected in 1.7.7, the [1.7.7 beta version release note](https://build.prestashop.com/news/prestashop-1-7-7-0-beta-release/) is the perfect article for you to read!
+
```
Comment (jolelievre) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md:75]: Add some infos about the tests here maybe? I don't know how to present it but here are the statistics:

```
Tests in 1.7.7:
UI (sanity + functional): 4307 (whole new campaigns)
Integration:
  - PHPUnit 55 tests (+3)
  - Behat 518 scenarios, 8838 steps (+297 scenarios, +6133 steps)
Unit: 1998 tests (+31)
```

And @SimonGrn suggested to integrate those two links if people want more info about UI tests:
- https://build.prestashop.com/news/the-new-prestashop-test-framework/
- https://build.prestashop.com/news/how-to-write-a-new-ui-test/

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
+---
+layout: post
+title:  "PrestaShop 1.7.7.0 RC1 Is Ready To Be Tested!"
+subtitle: "A month after the beta, here it is: the first release candidate for PrestaShop 1.7.7.0 is now ready for you to test!"
```
Comment (matks) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md]: Probably outdated 😢

File: news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md (hunk 11 above)
Comment (matks) [reply news/_posts/releases/2020-10-22-prestashop-1-7-7-0-rc-release.md]: Yes better rename it

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+date:   2020-11-03 15:00:00
```
[the remark this answers is not in the document]
Comment (matks) [reply news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md:5]: Duplicate with @jolelievre  => ignore my comment

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md (hunk 13 above)
Comment (matks) [reply news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md]: Duplicate with @jolelievre  => ignore my comment

**Review by matks — COMMENTED** [review]
Excluding necessary date changes, I'm OK with this release note 👍

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (jolelievre) [reply news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md:5]: You're right it's actually TWO months ^^

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 15]
```diff
@@ -57,19 +57,60 @@ If you find regressions compared to the 1.7.6 version, please open an issue and
...
-This brings us to more than 1300 merged pull requests for the 1.7.7.0 milestone, it is definitely our biggest yet.
+151 pull requests have been merged since the beta 2 release and 30+ issues have been fixed. Read the [Changelog](https://github.com/PrestaShop/PrestaShop/releases) for details. 
+This brings us to more than 1400 merged pull requests for the 1.7.7.0 milestone, it is definitely our biggest yet.
 
 If you are looking for more details about all changes and new features expected in 1.7.7, the [1.7.7 beta version release note](https://build.prestashop.com/news/prestashop-1-7-7-0-beta-release/) is the perfect article for you to read!
 
+## Automated Tests
+
+The 1.7.7 version is by far the most tested version ever released by PrestaShop, this ensures its stability and prevents future regressions that could be created by future developments.
+Here are a bit of numbers about the PrestaShop tests (and the difference with previous version 1.7.6):
```
Comment (eternoendless) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md]:
```suggestion
Here are some numbers about PrestaShop's automatic tests (and the difference with our previous minor release, 1.7.6):
```

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 16]
```diff
@@ -57,19 +57,60 @@ If you find regressions compared to the 1.7.6 version, please open an issue and
...
+
+- UI Puppeteer (sanity + functional): 4307 (whole new campaigns)
```
Comment (eternoendless) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md]:
```suggestion
- UI tests (sanity + functional): 4307 (whole new campaigns)
```

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 17]
```diff
@@ -57,19 +57,60 @@ If you find regressions compared to the 1.7.6 version, please open an issue and
...
+- Integration:
+  - PHPUnit 55 tests (+3)
+  - Behat 518 scenarios, 8838 steps (+297 scenarios, +6133 steps)
```
Comment (eternoendless) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md]:
```suggestion
- Integration: 55 tests (+3)
- Behavior: 518 scenarios, 8838 steps (+297 scenarios, +6133 steps)
```

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+icon: icon-leaf
+tags:
+ - 1.7.7
+ - version
+ - rc
+ - minor
+ - releases
+---
+
+We are really happy to announce that we are close to releasing 1.7.7.0, the release candidate is the last step before releasing the final version.
```
Comment (Progi1984) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md]:
```suggestion
We are really happy to announce that we are close to release 1.7.7.0, the release candidate is the last step before releasing the final version. 
```

File: news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md [hunk 19]
```diff
@@ -57,19 +57,60 @@ If you find regressions compared to the 1.7.6 version, please open an issue and
...
+  - PHPUnit 55 tests (+3)
+  - Behat 518 scenarios, 8838 steps (+297 scenarios, +6133 steps)
+- Unit: 1998 tests (+31)
+
+The UI tests, especially, have been completely redesigned. If you want more information about this please read these two articles:
+- [The (New) PrestaShop Test Framework](https://build.prestashop.com/news/the-new-prestashop-test-framework/)
+- [Automated UI Tests Tutorial](https://build.prestashop.com/news/how-to-write-a-new-ui-test/)
+
+## Known issues
+
```
Comment (eternoendless) [inline news/_posts/releases/2020-11-03-prestashop-1-7-7-0-rc-release.md:91]:
```suggestion

The following issues are not considered edge cases and won't block the release of a stable version. They will be fixed in the next patch version.

```

**Comment by eternoendless** [comment]
Thanks everyone!
